### PR TITLE
add functional clear button try

### PR DESCRIPTION
**Pull request by JetShen**
Fix: [issue ](https://github.com/pavo-etc/minecraftle/issues/11)

https://github.com/pavo-etc/minecraftle/assets/62623196/06aa1871-e7b9-4ad0-b723-faeba7fda557

### CODE REVIEW

**Review by TheVanadium — CHANGES_REQUESTED** [review]
Fix grammar on code doc - lines 332 and 333

**Review by TheVanadium — CHANGES_REQUESTED** [review]
I think it would look better aesthetically and intuitively if the guess counter was on the bottom. That way, all the crafting table-interaction tools are grouped, instead of divided by the counter.

File: app/static/js/init.js [hunk 1]
```diff
@@ -328,6 +328,29 @@ function addNewCraftingTable() {
   document.getElementById("guesses").appendChild(newTable);
 }
 
+/**
+ * Obtanin the length of the crafting table array
+ * Clear the lats crafting table
+ */
+function clearTable() {
+  let lastTable = craftingTables.length - 1;
```
Comment (TheVanadium) [inline app/static/js/init.js:336]: `currentTable` might be a better name here

**Comment by JetShen** [comment]
> I think it would look better aesthetically and intuitively if the guess counter was on the bottom. That way, all the crafting table-interaction tools are grouped, instead of divided by the counter. However, it's up to @pavo-etc

You mean like this?

image

**Comment by pavo-etc** [comment]
I think a better design would be to have the guess counter and clear button on the same line, but on opposite sides.

File: app/static/js/init.js [hunk 2]
```diff
@@ -328,6 +328,29 @@ function addNewCraftingTable() {
...
+  let solutionSlot = solution.children[0];
+  solutionSlot.removeAttribute("style");
+
+  for (let row = 0; row < craftingTables[lastTable].length; row++) {
+    for (let col = 0; col < craftingTables[lastTable][row].length; col++) {
+      craftingTables[lastTable][row][col] = null;
+
+      let slotId = row * 3 + col;
+      let slot = table.children[slotId].children[0];
+      slot.removeAttribute("style");
```
Comment (TheVanadium) [inline app/static/js/init.js:349]: This should use the `setSlotBackground` function for consistency with other functions

**Comment by TheVanadium** [comment]
> > I think it would look better aesthetically and intuitively if the guess counter was on the bottom. That way, all the crafting table-interaction tools are grouped, instead of divided by the counter. However, it's up to @pavo-etc
> 
> You mean like this?
> 
> image

Yeah I like that, but go with @pavo-etc since he's here

File: app/static/js/init.js [hunk 3]
```diff
@@ -328,6 +328,29 @@ function addNewCraftingTable() {
   document.getElementById("guesses").appendChild(newTable);
 }
 
+/**
+ * Obtanin the length of the crafting table array
+ * Clear the lats crafting table
```
[the remark this answers is not in the document]
Comment (JetShen) [reply app/static/js/init.js:333]: Could you give me an idea of how I would do the message for documentation?

**Comment by JetShen** [comment]
> I think a better design would be to have the guess counter and clear button on the same line, but on opposite sides.
@TheVanadium 
do you like it this way?

image

**Comment by TheVanadium** [comment]
> > I think a better design would be to have the guess counter and clear button on the same line, but on opposite sides.
> > @TheVanadium
> > do you like it this way?
> 
> image

Personally I prefer the `Clear Table` button on the left because the left is the side of the crafting table the user interacts with. On the right, the average mouse path is diagonal, which feel strange. However, ultimately up to @pavo-etc

**Comment by JetShen** [comment]
I forgot to change this part 💀 

image

**Comment by TheVanadium** [comment]
Can you figure out a way to disable the function when the game is over?

**Comment by JetShen** [comment]
> Can you figure out a way to disable the function when the game is over? 

Do you mean that the button disappears and the attempts remain in the center?

**Comment by TheVanadium** [comment]
I think preferably the button click should just do nothing. It should be something like a global variable that tracks whether or not the game is over yet, but I don't know if one exists so far.

**Comment by JetShen** [comment]
there is no such variable, what could be done is to disable/remove the button in the gain/loss functions.

**Review by TheVanadium — CHANGES_REQUESTED** [review]
`clearbutton` should be `gameIsFinished`, and `gameIsFinished` should be false by default, as the game starts unfinished, then set to true in the `winner()` and `loser()` functions.

**Pull request closed without merging**